src/components/RightSideDrawer.jsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+
+import Brightness4Icon from "@mui/icons-material/Brightness4";
+import Brightness7Icon from "@mui/icons-material/Brightness7";
+import LanguageIcon from "@mui/icons-material/Language";
+import MenuIcon from "@mui/icons-material/Menu";
+import PaletteIcon from "@mui/icons-material/Palette";
+import SettingsIcon from "@mui/icons-material/Settings";
+import { Box, List, ListItem, ListItemIcon, ListItemText, Divider, Select, MenuItem } from "@mui/material";
+import Avatar from "@mui/material/Avatar";
+import Drawer from "@mui/material/Drawer";
+import IconButton from "@mui/material/IconButton";
+import PropTypes from "prop-types";
+
+const RightSideDrawer = ({ themeMode, toggleThemeMode }) => {
+  const [open, setOpen] = React.useState(false);
+  const [isLoggedIn, setIsLoggedIn] = React.useState(false); // Assuming user login state
+  const [selectedColor, setSelectedColor] = React.useState("primary"); // 'primary' or 'secondary'
+  const [language, setLanguage] = React.useState("English"); // Default language
+  const languageList = ["English", "Spanish", "French", "German", "Italian"]; // Example language list
+
+  const handleToggleDrawer = () => {
+    setOpen(!open);
+  };
+
+  const handleColorChange = (event) => {
+    setSelectedColor(event.target.value);
+  };
+
+  const handleLanguageChange = (event) => {
+    setLanguage(event.target.value);
+  };
+
+  const handleLogout = () => {
+    // Perform logout action
+    setIsLoggedIn(false);
+  };
+
+  const renderAvatar = () => {
+    if (isLoggedIn) {
+      // Assuming user data is available in context
+      return (
+        <Avatar sx={{ bgcolor: "#3f51b5" }}>AB</Avatar> // Placeholder initials and background color
+      );
+    }
+    return <SettingsIcon />;
+  };
+
+  return (
+    <div>
+      <IconButton color="inherit" aria-label="open drawer" edge="end" onClick={handleToggleDrawer} sx={{ mr: 2 }}>
+        <MenuIcon />
+      </IconButton>
+      <Drawer
+        anchor="right" // Position the drawer on the right side
+        open={open}
+        onClose={handleToggleDrawer}
+        sx={{ width: "60vw" }} // Set the width of the drawer as 60% of the viewport width
+      >
+        <Box sx={{ width: "100%" }}>
+          <List>
+            <ListItem button onClick={toggleThemeMode}>
+              <ListItemIcon>{themeMode === "light" ? <Brightness4Icon /> : <Brightness7Icon />}</ListItemIcon>
+              <ListItemText primary="Theme Mode" />
+            </ListItem>
+            <ListItem>
+              <ListItemIcon>
+                <PaletteIcon />
+              </ListItemIcon>
+              <ListItemText primary="Theme Color" />
+              <Select value={selectedColor} onChange={handleColorChange}>
```

Comment:
,  create  Selector Component(reusable component)   , both  theme and language  use same component by passing props